src/plugProjectNishimuraU/DangoMushi.cpp

```diff
@@ -629,118 +552,43 @@ void Obj::setInitialSetting(EnemyInitialParamBase*) { }
  * Address:	802FC4F4
  * Size:	000130
  */
-void Obj::onInit(CreatureInitArg*)
+void Obj::onInit(CreatureInitArg* arg)
 {
-	/*
-	stwu     r1, -0x10(r1)
-	mflr     r0
-	stw      r0, 0x14(r1)
-	stw      r31, 0xc(r1)
-	mr       r31, r3
-	bl       onInit__Q24Game9EnemyBaseFPQ24Game15CreatureInitArg
-	lwz      r3, 0x1e0(r31)
-	li       r4, 0
-	lfs      f0, lbl_8051D3A8@sda21(r2)
-	li       r0, -1
-	ori      r5, r3, 1
-	mr       r3, r31
-	stw      r5, 0x1e0(r31)
-	lwz      r5, 0x1e0(r31)
-	rlwinm   r5, r5, 0, 0x1a, 0x18
-	stw      r5, 0x1e0(r31)
-	stb      r4, 0x2c0(r31)
-	stfs     f0, 0x2e4(r31)
-	stfs     f0, 0x2e0(r31)
-	stfs     f0, 0x2dc(r31)
-	stfs     f0, 0x2c4(r31)
-	stw      r0, 0x2cc(r31)
-	stfs     f0, 0x2c8(r31)
-	bl       setupCollision__Q34Game10DangoMushi3ObjFv
-	mr       r3, r31
-	li       r4, 0
-	bl       setBodyCollision__Q34Game10DangoMushi3ObjFb
-	li       r0, 0
-	mr       r3, r31
-	stb      r0, 0x2c1(r31)
-	li       r4, 0
-	bl       resetMapCollisionSize__Q34Game10DangoMushi3ObjFb
-	li       r0, 0
-	mr       r3, r31
-	stb      r0, 0x2c3(r31)
-	bl       resetBossAppearBGM__Q34Game10DangoMushi3ObjFv
-	mr       r3, r31
-	bl       setupEffect__Q34Game10DangoMushi3ObjFv
-	lwz      r3, shadowMgr__4Game@sda21(r13)
-	mr       r4, r31
-	bl       delShadow__Q24Game9ShadowMgrFPQ24Game8Creature
-	lwz      r3, gameSystem__4Game@sda21(r13)
-	cmplwi   r3, 0
-	beq      lbl_802FC5D4
-	lwz      r0, 0x44(r3)
-	cmpwi    r0, 4
-	bne      lbl_802FC5D4
-	lwz      r3, 0x2bc(r31)
-	mr       r4, r31
-	li       r5, 2
-	li       r6, 0
-	lwz      r12, 0(r3)
-	lwz      r12, 0xc(r12)
-	mtctr    r12
-	bctrl
-	b        lbl_802FC5F4
-
-lbl_802FC5D4:
-	lwz      r3, 0x2bc(r31)
-	mr       r4, r31
-	li       r5, 1
-	li       r6, 0
-	lwz      r12, 0(r3)
-	lwz      r12, 0xc(r12)
-	mtctr    r12
-	bctrl
-
-lbl_802FC5F4:
-	lwz      r3, 0x2f0(r31)
-	lwz      r4, 0x180(r31)
-	lwz      r12, 0(r3)
-	lwz      r4, 0x44(r4)
-	lwz      r12, 8(r12)
-	mtctr    r12
-	bctrl
-	lwz      r0, 0x14(r1)
-	lwz      r31, 0xc(r1)
-	mtlr     r0
-	addi     r1, r1, 0x10
-	blr
-	*/
+	EnemyBase::onInit(arg);
+	mEvents.mFlags->typeView |= 1;
```

Comment:
I'd recommend looking through the EnemyBase.h header (near the top) for the events enum, and using the `enableEvent`, `disableEvent` and `isEvent` inlines with those to turn stuff on and off, since they come up a lot.